fields.yml

```diff
@@ -745,6 +745,79 @@
           description: >
             Operating system architecture.
     
+    - name: http
+      title: HTTP
+      group: 2
+      description: >
+        Fields related to HTTP activity.
+      type: group
+      fields:
+    
+        - name: request.method
+          level: extended
+          type: keyword
+          description: >
+            Http request method.
+          example: GET, POST, PUT
+    
+        - name: request.referrer
+          level: extended
+          type: keyword
+          description: >
+            Referrer for this HTTP request.
+          example: https://blog.example.com/
+    
+        - name: response.status_code
+          level: extended
+          type: long
+          description: >
+            Http response status code.
+          example: 404
+    
+        - name: response.body
+          level: extended
+          type: keyword
+          description: >
+            The full http response body.
+          example: Hello world
+    
+        - name: version
+          level: extended
+          type: keyword
+          description: >
+            Http version.
+          example: 1.1
+    
+        # Metrics
+        - name: request.bytes.total
+          level: extended
+          type: long
+          description: >
+            Size in bytes of the request body and headers combined.
+          example: 1437
+    
+        - name: request.bytes.body
+          level: extended
+          type: long
+          description: >
+            Size in bytes of the request body.
+          example: 887
+    
+        - name: response.bytes.total
+          level: extended
+          type: long
+          description: >
+            Size in bytes of the response body and headers combined.
+          example: 1437
+    
+        - name: response.bytes.body
```

Comment:
I like the grouping this enables: `header.bytes`, `body.bytes`. Which eventually leaves room for other metadata, if we ever want to track `body.encoding` & such.

However do we use `http.[request|response].body` anywhere? Would moving that to `....body.content` be another "concrete field" => object transition, like `source`?

Reply:
It would also allow adding something like `body.hash` if the body should be compared but not the full body stored.

We use `http.request.body` today in Metricbeat in the http module. So an alias would not be possible but I think it's fine in this case as it is with `source`.